pymc3/distributions/multivariate.py

```diff
@@ -1003,17 +1107,15 @@ def logp(self, x):
         -------
         TensorVariable
         """
-        n = self.n
-        eta = self.eta
 
-        diag_idxs = self.diag_idxs
+        diag_idxs = at.cumsum(at.arange(1, n + 1)) - 1
         cumsum = at.cumsum(x ** 2)
         variance = at.zeros(n)
         variance = at.inc_subtensor(variance[0], x[0] ** 2)
         variance = at.inc_subtensor(variance[1:], cumsum[diag_idxs[1:]] - cumsum[diag_idxs[:-1]])
         sd_vals = at.sqrt(variance)
 
-        logp_sd = self.sd_dist.logp(sd_vals).sum()
+        logp_sd = sd_dist.logp(sd_vals).sum()
```

Comment:
`sd_dist` becomes a `rv.out`(for the given distribution) here. What would be the right was to access it's `logp` method here ?

Reply:
You can call the _logp method on that `dist`. Check the ZeroInflatedBinomial logp for an example.

Reply:
Something like `_logp(sd_dist.owner.op, {}, sd_vals, *sd_dist.owner.inputs)`